src/lucky/request.cr

```diff
@@ -0,0 +1,87 @@
+require "http/client"
+
+class Lucky::Request
+  getter! url
+  getter! headers
+  getter! query_params
+  getter! response
+
+  @url : String? = nil
+  @headers : HTTP::Headers? = nil
+  @query_params : String? = nil
+  @response : HTTP::Client::Response? = nil
+
+  def get(path : String, as user = nil, headers : HTTP::Headers? = nil)
+    if user
+      headers ||= HTTP::Headers.new
+      headers.add "Authorization", user.generate_token
+    end
```

Comment:
is it worth making these 4 lines into a helper function?

Reply:
Thanks for pointing that out!